File: exercises/ruby_more_topics/easy_01/03.rb
# Exercise 03

# Write a method that takes a sorted array of integers as an argument, 
# and returns an array that includes all of the missing integers (in order) 
# between the first and last elements of the argument.

# get every integer from the first to the last in input array

# delete integer in output array if it is in input array



def missing(input_array)
  output_array = [*(input_array.first..input_array.last)]

  output_array.each_with_object([]) do |integer, output_array|
    output_array << integer unless input_array.include?(integer)
  end
end

p missing([-3, -2, 1, 5]) == [-1, 0, 2, 3, 4]
p missing([1, 2, 3, 4]) == []
p missing([1, 5]) == [2, 3, 4]
p missing([6]) == []

File: exercises/js_small_problems/easy_02/03.js
// Exercise 03

// Write a function that takes one argument, a positive integer, 
// and returns a string of alternating '1's and '0's, 
// always starting with a '1'. The length of the string should match the given integer.

function stringy(size) {
  let output = '';
  for (let i = 0; i < size; i += 1) {
    output += i % 2 === 0 ? '1' : '0';
  }
  return output;
}

console.log(stringy(6));    // "101010"
console.log(stringy(9));    // "101010101"
console.log(stringy(4));    // "1010"
console.log(stringy(7));    // "1010101"